vpr/src/place/noc_place_utils.cpp

```diff
@@ -552,19 +546,47 @@ double calculate_traffic_flow_aggregate_bandwidth_cost(const std::vector<NocLink
 std::pair<double, double> calculate_traffic_flow_latency_cost(const std::vector<NocLinkId>& traffic_flow_route,
                                                               const NocStorage& noc_model,
                                                               const t_noc_traffic_flow& traffic_flow_info) {
-    // there will always be one more router than links in a traffic flow
-    int num_of_links_in_traffic_flow = traffic_flow_route.size();
-    int num_of_routers_in_traffic_flow = num_of_links_in_traffic_flow + 1;
-    double max_latency = traffic_flow_info.max_traffic_flow_latency;
 
-    // latencies of the noc
-    double noc_link_latency = noc_model.get_noc_link_latency();
-    double noc_router_latency = noc_model.get_noc_router_latency();
+    double noc_link_latency_component = 0.0;
+    if (noc_model.get_detailed_link_latency()) {
+        for (const NocLink& link : noc_model.get_noc_links(traffic_flow_route)) {
+            double link_latency = link.get_latency();
+            noc_link_latency_component += link_latency;
+        }
+    } else {
+        auto num_of_links_in_traffic_flow = (double)traffic_flow_route.size();
+        double noc_link_latency = noc_model.get_noc_link_latency();
+        noc_link_latency_component = noc_link_latency * num_of_links_in_traffic_flow;
```

Comment:
If you stored the latency as part of the routing computation, then you could avoid this if statement / two methods. Consider if that is easier to understand (gets rid of a state control variable).

Also can get rid of the detailed bandwidth variable. (if it controls nothing, I'd get rid of it).

Reply:
I think we should keep all NoC-related cost compuation code in a single file. We can't move congestion computation into the routing algorithm. So, moving the latency and aggregate bandwidth computation to the routing algorithm splits the NoC-realted cost calcuation code across files, which is not ideal.

I also think that if we move aggregate bandwidth and latency calculation from here to routing algrotihms, we need to implement the calculation process multiple times as NoCRouting::route_flow(...) method is pure virtual.

Reply:
Valid points.  OK, let's leave as is.